src/middleware/minitrace_grpc.rs

```diff
@@ -76,56 +76,52 @@ where
     }
 
     /// https://docs.datadoghq.com/tracing/trace_collection/trace_context_propagation/#custom-header-formats
-    fn call(&mut self, req: tonic::codegen::http::Request<Body>) -> Self::Future {
-        // This is necessary because tonic internally uses `tower::buffer::Buffer`.
-        // See https://github.com/tower-rs/tower/issues/547#issuecomment-767629149
-        // for details on why this is necessary
-        let clone = self.inner.clone();
-        let mut inner = std::mem::replace(&mut self.inner, clone);
```

Comment:
with the removal of this workaround, slight change was needed to do `let future = self.inner.call(req).in_span(root_span);` outside `aysnc move` and await it inside